submit_cgap/tests/test_submission.py

```diff
@@ -2202,3 +2253,144 @@ def test_running_on_windows_native():
         os_name, is_windows = pair
         with mock.patch.object(os, "name", os_name):
             assert running_on_windows_native() is is_windows
+
+
+@pytest.mark.parametrize(
+    "directory,file_name,recursive,glob_results,expected_file_path,expected_msg",
+    [
+        ("foo", "bar", False, [], "foo/bar", False),
+        ("foo", "bar", True, [], "foo/bar", False),
+        ("foo", "bar", False, ["foo/bar"], "foo/bar", False),
+        ("foo", "bar", False, ["foo/bar", "fu/foo/bar"], None, True),
+    ]
+)
+def test_search_for_file(
+    directory, file_name, recursive, glob_results, expected_file_path, expected_msg
+):
+    """Test output file path +/- error message dependent on file search
+    via glob.
+    """
+    with mock.patch(
+        "submit_cgap.submission.glob.glob", return_value=glob_results
+    ) as mocked_glob:
+        file_path_found, error_msg = search_for_file(directory, file_name, recursive)
+        mocked_glob.assert_called_once_with(
+            directory + "/" + file_name, recursive=recursive
+        )
+        assert file_path_found == expected_file_path
+        if expected_msg:
+            assert error_msg.startswith(
+                f"No upload attempted for file {file_name}"
+            )
+
```

Comment:
Seems like there should be an `slse` that checks that does, perhaps something like:
```
assert not error_msg, "Got error message where not expected."
```